packages/url_launcher/url_launcher_windows/windows/url_launcher_plugin.cpp

```diff
@@ -120,29 +108,48 @@ void UrlLauncherPlugin::HandleMethodCall(
       return;
     }
 
-    bool can_launch = false;
-    size_t separator_location = url.find(":");
-    if (separator_location != std::string::npos) {
-      std::wstring scheme = Utf16FromUtf8(url.substr(0, separator_location));
-      HKEY key = nullptr;
-      if (::RegOpenKeyEx(HKEY_CLASSES_ROOT, scheme.c_str(), 0, KEY_QUERY_VALUE,
-                         &key) == ERROR_SUCCESS) {
-        can_launch = ::RegQueryValueEx(key, L"URL Protocol", nullptr, nullptr,
-                                       nullptr, nullptr) == ERROR_SUCCESS;
-        ::RegCloseKey(key);
-      }
-    }
+    bool can_launch = CanLaunchUrl(url);
     result->Success(EncodableValue(can_launch));
   } else {
     result->NotImplemented();
   }
 }
 
-}  // namespace
+bool UrlLauncherPlugin::CanLaunchUrl(const std::string& url) {
+  size_t separator_location = url.find(":");
```

Comment:
Off-topic for this patch, but I bet there's some Win32 API for parsing URLs out there. If the Dart side of this is already using the `Url` class it's much less of an issue; otherwise, : can appear as the port separator etc., or just as part of some garbage string the person passed in here. I suppse the net result would mostly be them shooting themselves in their own foot regardless unless they're writing an app to poke at a very specific and maybe not-particularly-valuable to know subset of users' registry keys.

Reply:
> If the Dart side of this is already using the `Url` class it's much less of an issue

The Dart side of the API is kind of a hot mess (https://github.com/flutter/flutter/issues/79043), and one of the things I dislike most about it is that it takes a `String`. Someday I want to get back to that proposal and do the overhaul.

> I suppse the net result would mostly be them shooting themselves in their own foot regardless

Yes, and at least unlike some of the other issues (like including unencoded spaces, which people do all the time) trying to pass a "URL" that doesn't actually have a scheme should completely fail on every platform, so be very easy to notice during development.